src/overlap_check/overlap.cpp

```diff
@@ -0,0 +1,184 @@
+                                     const Range& all_vols,
+                                     CartVect loc,
+                                     CartVect dir,
+                                     OverlapMap& overlap_map) {
+  ErrorCode rval;
+
+  GeomTopoTool* GTT = GQT->gttool();
+  std::set<int> vols_found;
+  double bump = 1E-9;
+
+  // move the point slightly off the vertex
+  loc += dir * bump;
+
+  for (const auto& vol : all_vols) {
+    int result = 0;
+    rval = GQT->point_in_volume(vol, loc.array(), result, dir.array());
+    MB_CHK_SET_ERR(rval, "Failed point in volume for Vol with id "
+                   << GTT->global_id(vol)
+                   << " at location " << loc);
+
+    if (result == 1) {
+      vols_found.insert(GTT->global_id(vol));
+    }
+  }
+
+  if (vols_found.size() > 1) {
+    overlap_map[vols_found] = loc;
+  }
+
+  // move the point slightly off the vertex
+  dir *= -1;
+  loc += dir * 2.0 * bump;
+  vols_found.clear();
+
+  for (const auto& vol : all_vols) {
+    int result = 0;
+    rval = GQT->point_in_volume(vol, loc.array(), result, dir.array());
+    MB_CHK_SET_ERR(rval, "Failed point in volume for Vol with id "
+                   << GTT->global_id(vol)
+                   << " at location " << loc);
+
+    if (result == 1) {
+      vols_found.insert(GTT->global_id(vol));
+    }
+  }
+
+  if (vols_found.size() > 1) {
+    overlap_map[vols_found] = loc;
+  }
+
+  return MB_SUCCESS;
+}
+
+ErrorCode
+check_instance_for_overlaps(std::shared_ptr<Interface> MBI,
+                            OverlapMap& overlap_map,
+                            int pnts_per_edge) {
+
+  std::shared_ptr<GeomTopoTool> GTT(new GeomTopoTool(MBI.get()));
+  std::shared_ptr<GeomQueryTool> GQT(new GeomQueryTool(GTT.get()));
+
+  ErrorCode rval = GTT->find_geomsets();
+  MB_CHK_SET_ERR(rval, "Failed to detect geometry sets");
+
+  rval = GTT->construct_obb_trees();
+  MB_CHK_SET_ERR(rval, "Failed to build OBB trees");
+
+  Range all_verts;
+  rval = MBI->get_entities_by_type(0, MBVERTEX, all_verts);
+  MB_CHK_SET_ERR(rval, "Failed to get all vertices");
+
+  Range all_tris;
+  rval = MBI->get_entities_by_type(0, MBTRI, all_tris);
+  MB_CHK_SET_ERR(rval, "Failed to get all triangles");
+
+  Range all_edges;
+  rval = MBI->get_adjacencies(all_tris, 1, true, all_edges, Interface::UNION);
+  MB_CHK_SET_ERR(rval, "Failed to get triangle edges");
+
+  Range all_vols;
+  rval = GTT->get_gsets_by_dimension(3, all_vols);
+  MB_CHK_SET_ERR(rval, "Failed to get volumes from GTT");
+
+  // number of locations we'll be checking
+  int num_locations = all_verts.size() + pnts_per_edge * all_edges.size();
+  int num_checked = 1;
+
+  CartVect dir(rand(), rand(), rand());
+  dir.normalize();
+
+  ProgressBar prog_bar;
+
+  // first check all triangle vertex locations
+  #pragma omp parallel shared(overlap_map, num_checked)
+  {
+    #pragma omp for schedule(auto)
+    for (int i = 0; i < all_verts.size(); i++) {
```

Comment:
`int` should be replaced with `size_t` here. At a minimum this will remove a compiler warning about mismatched signed comparison against unsigned type, but it will also guard against problems that are difficult to track down if this is ever built on a less conventional platform.